### ISSUE TITLE
pymc3 tries to sample model twice in simple code

### ISSUE THREAD

**Issue by sammosummo** [pymc-devs/pymc#4089]
Installation:

- macOS 10.15.6
- installed via `conda-forge`
- Python 3.8.5
- pymc3 3.9.3

This trivial example tries to sample the same model twice:

```
import pymc3 as pm
with pm.Model() as m: x = pm.Normal("x")
pm.sample(model=m)
```

Here is the output:

```
Auto-assigning NUTS sampler...
Initializing NUTS using jitter+adapt_diag...
Multiprocess sampling (3 chains in 3 jobs)
NUTS: [x]
Auto-assigning NUTS sampler...
Initializing NUTS using jitter+adapt_diag...
Multiprocess sampling (3 chains in 3 jobs)
NUTS: [x]
Traceback (most recent call last):
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/forkserver.py", line 280, in main
    code = _serve_one(child_r, fds,
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/forkserver.py", line 319, in _serve_one
    code = spawn._main(child_r, parent_sentinel)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/spawn.py", line 125, in _main
    prepare(preparation_data)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/samuelrobertmathias/PycharmProjects/ForgettingPitchPublic/_test.py", line 5, in <module>
    pm.sample(model=m, progressbar=True)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/sampling.py", line 545, in sample
    trace = _mp_sample(**sample_args, **parallel_args)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/sampling.py", line 1470, in _mp_sample
    sampler = ps.ParallelSampler(
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/parallel_sampling.py", line 442, in __init__
    self._samplers = [
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/parallel_sampling.py", line 443, in <listcomp>
    ProcessAdapter(
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/parallel_sampling.py", line 294, in __init__
    self._process.start()
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/context.py", line 291, in _Popen
    return Popen(process_obj)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/popen_forkserver.py", line 35, in __init__
    super().__init__(process_obj)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/popen_forkserver.py", line 42, in _launch
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError: 
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
Traceback (most recent call last):
  File "/Users/samuelrobertmathias/PycharmProjects/ForgettingPitchPublic/_test.py", line 5, in <module>
    pm.sample(model=m, progressbar=True)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/sampling.py", line 545, in sample
    trace = _mp_sample(**sample_args, **parallel_args)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/sampling.py", line 1470, in _mp_sample
    sampler = ps.ParallelSampler(
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/parallel_sampling.py", line 442, in __init__
    self._samplers = [
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/parallel_sampling.py", line 443, in <listcomp>
    ProcessAdapter(
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/site-packages/pymc3/parallel_sampling.py", line 294, in __init__
    self._process.start()
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/context.py", line 291, in _Popen
    return Popen(process_obj)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/popen_forkserver.py", line 35, in __init__
    super().__init__(process_obj)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Users/samuelrobertmathias/miniconda3/envs/ForgettingPitchPublic/lib/python3.8/multiprocessing/popen_forkserver.py", line 58, in _launch
    f.write(buf.getbuffer())
BrokenPipeError: [Errno 32] Broken pipe

Process finished with exit code 1
```

**Comment by fonnesbeck** [comment]
On some platforms, the chains are run sequentially by default. Try adding a `chains=3` argument to sample and see if that fixes it.

**Comment by thomasaarholt** [comment]
I can reproduce this on my M1 Macbook Pro on Python 3.9. I can bypass it by passing `cores=1` to `pm.sample()`, but it is a shame that default behaviour crashes it.

Duplicate of this issue in #4167.

**Comment by fonnesbeck** [comment]
On Colab (the only place I have seen this behavior) passing `cores=2` will sample in parallel. `cores=1` for multiple chains will result in sequential sampling.

**Comment by fonnesbeck** [comment]
Closing as duplicate
